qiskit/transpiler/passes/optimization/normalize_rx_angle.py

```diff
@@ -0,0 +1,149 @@
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Performs three optimizations to reduce the number of pulse calibrations for
+the single-pulse RX gates:
+Wrap RX Gate rotation angles into [0, pi] by sandwiching them with RZ gates.
+Convert RX(pi/2) to SX, and RX(pi) to X if the calibrations exist in the target.
+Quantize the RX rotation angles using a resolution provided by the user.
+"""
+
+import numpy as np
+
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.circuit.library.standard_gates import RXGate, RZGate, SXGate, XGate
+
+
+class NormalizeRXAngle(TransformationPass):
+    """Wrap RX Gate rotation angles into [0, pi] by sandwiching them with RZ gates.
+    This will help reduce the size of calibration data,
+    as we won't have to keep separate, phase-flipped calibrations for negative rotation angles.
+    Moreover, if the calibrations exist in the target, convert RX(pi/2) to SX, and RX(pi) to X.
+    This will allow us to exploit the more accurate, hardware-calibrated pulses.
+    Lastly, quantize the RX rotation angles using a resolution provided by the user.
+    """
+
+    def __init__(self, target=None, resolution_in_radian=0):
+        """NormalizeRXAngle initializer.
+
+        Args:
+            target (Target): The :class:`~.Target` representing the target backend.
+                If the target contains SX and X calibrations, this pass will replace the
+                corresponding RX gates with SX and X gates.
+            resolution_in_radian (float): Resolution for RX rotation angle quantization.
+                If set to zero, this pass won't modify the rotation angles in the given DAG.
+                (=Provides aribitary-angle RX)
+        """
+        super().__init__()
+        self.target = target
+        self.resolution_in_radian = resolution_in_radian
+        self.already_generated = {}
+
+    def quantize_angles(self, qubit, original_angle):
+        """Quantize the RX rotation angles using a resolution provided by the user.
+
+        Args:
+            qubit (Qubit): This will be the dict key to access the list of quantized rotation angles.
+            original_angle (float): Original rotation angle, before quantization.
+
+        Returns:
+            float: Quantized angle.
+        """
+
+        # check if there is already a calibration for a simliar angle
+        try:
+            angles = self.already_generated[qubit]  # 1d ndarray of already generated angles
+            quantized_angle = float(
+                angles[np.where(np.abs(angles - original_angle) < (self.resolution_in_radian / 2))]
```

Comment:
```suggestion
                angles[np.isclose(angles, original_angle, rtol=self.resolution_in_radian / 2))]
```
This might be simpler. You need to divide the resolution by 2 because you are considering ± half resolution?

Reply:
Thank you for the suggestion. Yes, that's why I divided the resolution by 2.

Reply:
I would like to use `atol=resolution/2` (absolute tolerance) instead of `rtol` (relative tolerance). The manual says `np.isclose(a,b)` evaluates `absolute(a - b) <= (atol + rtol * absolute(b))`. https://numpy.org/doc/stable/reference/generated/numpy.isclose.html